.github/workflows/redis-cluster.yml

```diff
@@ -0,0 +1,56 @@
+name: Test Redis Cluster
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+# Cancel in progress workflows on pull_requests.
+# https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  redis_cluster:
+    runs-on: ubuntu-latest
+    name: redis backend test
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        pg-version: ['14']
+
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          # Avoid codecov error message related to SHA resolution:
+          # https://github.com/codecov/codecov-bash/blob/7100762afbc822b91806a6574658129fe0d23a7d/codecov#L891
+          fetch-depth: '2'
+
+      - name: Setup sentry env
+        uses: ./.github/actions/setup-sentry
+        id: setup
+        with:
+          redis_cluster: true
+          kafka: true
+          snuba: true
+          symbolicator: true
+          # Right now, we run so few bigtable related tests that the
+          # overhead of running bigtable in all backend tests
+          # is way smaller than the time it would take to run in its own job.
+          bigtable: true
+          pg-version: ${{ matrix.pg-version }}
+
+      - name: Run redis cluster backend test
+        run: |
+          make test-redis-cluster-ci
```

Comment:
I think you can use `PYTEST_ADDOPTS="$PYTEST_ADDOPTS -m redis_cluster" make test-python-ci`

or whatever the original target is -- this makes it so we don't  have to keep the testfilenames in sync

Reply:
I'm fine with replacing this, but I vote for keeping makefile change since it makes my job easier. Wdyt @asottile-sentry?

Reply:
ah I'm surprised people actually use the makefile to run things... there's so many tests that I find it mostly useless locally

`pytest tests/sentry -m redis_cluster` probably gets you most of what you want I would imagine? and is short enough?